### ISSUE TITLE
FESOM 2.0 standalone doesn't compile on ECMWF-atos

### ISSUE THREAD

**Issue by tsemmler05** [esm-tools/esm_tools#1207]
Trying to install and compile FESOM 2.0 standalone with esm-tools version 6.37.2 (branch ecmwf-atos-new), I'm getting the following error message when typing esm_master install-fesom-2.0:

-- Looking for Fortran NCCRE
-- Looking for Fortran NCCRE - not found
-- Looking for nc__create
-- Looking for nc__create - not found
-- Configuring done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files.

However, when trying to install and compile AWI-CM3 version 3.1 (esm_master install-awicm3-v3.1), I'm getting the same "not found" messages but the model will compile successfully (using the same esm-tools version and branch). It won't run more than a few time steps before blowing up, though (for both TCO95-CORE2 and TCO319-DART).

Since I need to run FESOM 2.0 standalone to see if the model blowup in the coupled version has to do with the oasis coupler or not: how can I circumvent the "not found" problem (and what are the Fortran NCCRE and nc__create modules for)?

**Comment by mandresm** [comment]
Hi @tsemmler05,

I have not reproduced the error yet, but I see already something that is weird: `ecmwf-atos-new` has departed very rencently from `release` and is just one commit. That means that you have probably manually copied all the new files for ECMWF-atos development into the new branch.

If that's the case, that's not an optimal practice because:
1. You delete the git history which is bad for keeping the history development itself and also because git commits into `release` is a metric to evaluate the work of the people that do a lot of development like me, so in a sense deleting my contributions in the form of commits and making them be your commits reduces my metric as developer (no big deal for me, but for other projects with people starting a career is not ideal)
2. You might make a mistake and forget to include some important file, which could explain the error.

There are some special cases in which you would still want to do this manually, but I very very rarely need to do it this way.

My recommendation is that you create a new branch departing from `ecmwf-atos` and merge `release` into that one. Of course there will be conflicts that you'll need to solve but they shouldn't be too difficult.

Please, let me know if you want to try that or whether you are convince your current approach is the correct one. If you want to try my suggested approach I am happy to help you with that via a webex call, so that you see how it is done.

Cheers,
Miguel

**Comment by mandresm** [comment]
If you want to speed up the process I can also do what I am suggesting myself, although it might be good that you learn how to do what I'm saying because is a pretty common git workflow. Let me know what's your preference :)

**Comment by tsemmler05** [comment]
The reason why I did everything from scratch was that running AWI-CM3.1 doesn't work anymore on ecmwf-atos and I wanted to do a fresh start. It worked in November / December 2023, then there was a system change at ECMWF, and now I'm struggling since a few months to get the code running at ECMWF. So, no matter if I use the old ecmwf-atos branch or the new one, there is the same problem. Of course, this does not directly have to do with the method to merge the original ecmwf-atos into release branch which I can see why this is superior to my approach. It is just that I desperately need a running AWI-CM3.1 version at ECMWF-atos, so that ECMWF user support can help me to fix the very slow runtime (as of November / December 2023, the code was consuming 3 times more computing power on ecmwf-atos compared to levante and obviously at the same time it ran 3 times slower in realtime when using the same amount of processors). But now, that I'm getting model blowups after a few time steps, ECMWF user support can't even start helping me with the bad computing performance.

**Comment by tsemmler05** [comment]
I could try to merge ecmwf-atos branch to release branch, though. There might be just a few extra changes to be applied because in the meantime, to try to cope with the ECMWF system changes, there have been some additional changes to ecmwf-atos branch, particularly fixing the compiler versions to the versions used in 2023 (without success, though).

**Comment by mandresm** [comment]
Since this seems urgent, I'm happy to do the merge myself. So let me know if you want that. Then we can start right away debugging and getting it back to running again.

**Comment by tsemmler05** [comment]
Yes, I guess in this case it would be good. I keep in mind the merging choice for the next time, though. Thanks for your help!

**Comment by mandresm** [comment]
Hi @tsemmler05,

I've already merged `release` into the old `ecmwf-atos`. To make sure all your new changes are also in that branch I have merged `ecmwf-atos-new` into `ecmwf-atos` (see drawing below for clarification). From now on, we can continue working in `ecmwf-atos` and I will soon remove the new branch.

![Screenshot 2024-08-02 at 08 01 08-1](https://github.com/user-attachments/assets/1da0c58e-71c4-48fc-aeb5-4569c6922e5c)

I will start now looking at the problem with the FESOM compilation
